### ISSUE TITLE
Internal compiler error for default struct with unknown field in async function

### ISSUE THREAD

**Issue by f2cognite** [rust-lang/rust#66554]
rustc panics when it tries to compile a struct in an async function with a unknown field and default expansion.

This is the smallest example I could find that would reproduce the problem:

```rust
pub struct Foo { }

fn bar(foo: Foo) { }

async fn run() { // async function
    bar(Foo {
        x: (), // unknown field in struct
        ..Default::default() // default expansion
    })
}
```

Even though this code isn't valid (due to the unknown field in the struct), I would have expected an error message like this (which is what I get if `run` is a sync function instead):

```
error[E0560]: struct `Foo` has no field named `x`
 --> test.rs:7:9
  |
8 |         x: (),
  |         ^ `Foo` does not have this field

error: aborting due to previous error

For more information about this error, try `rustc --explain E0560`.
```

However, in addition to the above error message I get an internal compiler error:

```
error[E0560]: struct `Foo` has no field named `x`
 --> test.rs:7:9
  |
7 |         x: (),
  |         ^ `Foo` does not have this field

error: internal compiler error: src/librustc/ty/context.rs:554: node_type: no type for node `expr Default::default (hir_id=HirId { owner: DefIndex(14), local_id: 8 })`

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:812:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.39.0 (4560ea788 2019-11-04) running on x86_64-apple-darwin

note: compiler flags: --crate-type dylib

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0560`.
```

## Meta

`rustc --version --verbose`: 

```
rustc 1.39.0 (4560ea788 2019-11-04)
binary: rustc
commit-hash: 4560ea788cb760f0a34127156c78e2552949f734
commit-date: 2019-11-04
host: x86_64-apple-darwin
release: 1.39.0
LLVM version: 9.0
```

Backtrace:

[backtrace.txt](https://github.com/rust-lang/rust/files/3865837/backtrace.txt)

**Comment by jonas-schievink** [comment]
This was already fixed (the fix is currently on nightly)
